### PR TITLE
Add a test to verify that a bug is fixed.

### LINKED ISSUE #27322: Generated code for Expression-tree instantiation crashes when it refers to ValueTuple<A,B>.Equals
**Version Used**: Various, including VS15.7.3 and VS15.8.0 preview2

**Steps to Reproduce**:
```C#
var tupleA = (1, 3);
var tupleB = (1, "123".Length);

Expression<Func<int>> ok1 = () => tupleA.Item1;
Expression<Func<int>> ok2 = () => tupleA.GetHashCode();
Expression<Func<Tuple<int, int>>> ok3 = () => tupleA.ToTuple();
Expression<Func<bool>> ok4 = () => Equals(tupleA, tupleB);
Expression<Func<int>> ok5 = () => Comparer<(int, int)>.Default.Compare(tupleA, tupleB);
ok1.Compile()(); //no crash.
ok2.Compile()(); //no crash.
ok3.Compile()(); //no crash.
ok4.Compile()(); //no crash.
ok5.Compile()(); //no crash.

Expression<Func<bool>> err1 = () => tupleA.Equals(tupleB);  //crash 1!
Expression<Func<int>> err2 = () => tupleA.CompareTo(tupleB);  //crash 2!
```
**Expected Behavior**:
The code runs without throwing.

**Actual Behavior**:

The above snippet crashes at line `crash 1` with ``System.ArgumentException : Cannot resolve method Boolean Equals(System.ValueTuple`2[System.Int32,System.Int32]) because the declaring type of the method handle System.ValueTuple`2[T1,T2] is generic. Explicitly provide the declaring type to GetMethodFromHandle.``

If you remove that line, it crashes at line `crash 2` with ``System.ArgumentException : Cannot resolve method Int32 CompareTo(System.ValueTuple`2[System.Int32,System.Int32]) because the declaring type of the method handle System.ValueTuple`2[T1,T2] is generic. Explicitly provide the declaring type to GetMethodFromHandle.``

------
Note (from @gafter):
This bug has been confirmed fixed in C# (https://github.com/dotnet/roslyn/pull/43518), but not in VB.  See discussion below.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #27322 (C# only)
VB fix depends on #20648

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using System.Linq;
 using System.Reflection.Metadata;
 using System.Text;
+using Castle.DynamicProxy.Generators.Emitters.SimpleAST;
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs:12]: Is this used?

**Comment by jcouv** [comment]
Consider adding the corresponding VB test and link it with https://github.com/dotnet/roslyn/issues/20648 so that the scenario doesn't get lost

**Comment by gafter** [comment]
> Consider adding the corresponding VB test and link it with #20648 so that the scenario doesn't get lost

I'm going to keep the bug open and add clarifying comments that the VB side is pending.

**Comment by gafter** [comment]
I did not intend this PR to fix anything.  I do not intend to expand its scope.

---
In reply to: [397487563](https://github.com/dotnet/roslyn/pull/43518#issuecomment-397487563) [](ancestors = 397487563)

**Pull request merged**